### ISSUE TITLE
LIKE '%Wert'” not working correct with ogr.ExecuteSQL and dialect = SQLITE

### ISSUE THREAD

**Issue by dpalmetzvbg** [OSGeo/gdal#490]
## Expected behavior and actual behavior.
I am trying to count features that have the text "Fehler" in a certain field, so i was using `WHERE rstatus LIKE '%Fehler'`. This only works, when using the OGRSQL-dialect, but returns None when using SQLITE. I want to use the SQLITE-dialect, because I am looping over several SQL-Strings, and some of them contain SUM(LENGTH)/1000, which only works in SQLITE, and not with OGRSQL. I checked the field-values, there are no trailing spaces. When using `trim(rstatus) like '%Fehler'` the statement also works with SQLITE dialect.
Further I was wondering, why the result is None, rather than a layer-object where GetFeatureCount = 0, when no records meet the criteria.

[OGR_Select.zip](https://github.com/OSGeo/gdal/files/1941286/OGR_Select.zip)

## Steps to reproduce the problem.

    import os

    try:
        from osgeo import ogr
        from osgeo import gdal
    except:
        sys.exit("Failed to import OGR-module. Please check the GDAL-Installation!")

    srcFile = r"D:\Didi\Tmp\OGR_Select\Export.shp"
    driver = ogr.GetDriverByName('ESRI Shapefile')
    destFileDS = driver.Open(srcFile, 0)
    sql = u"SELECT COUNT(*) as Anzahl from %s WHERE rstatus like '%%Fehler'" % (os.path.basename(srcFile)[:-4])

    #Works
    actLayer =  destFileDS.ExecuteSQL(statement=sql.encode('utf-8'), dialect='OGRSQL')
    actFeat = actLayer.GetNextFeature()
    destFileDS.ReleaseResultSet(actLayer)

    #Returns none
    actLayer =  destFileDS.ExecuteSQL(statement=sql.encode('utf-8'), dialect='SQLITE')
    actFeat = actLayer.GetNextFeature()
    destFileDS.ReleaseResultSet(actLayer)

## Operating system

Windows 7, 64 Bit
Python 2.7.5

## GDAL version and provenance

GDAL 2.1.3

**Comment by jratike80** [comment]
It seems that this happens only when the reference string begins with "%". I tested with this csv

id,rstatus
1,FehlerA
2,FehlerB
3,fehlerC 

This query finds all 3 features (I was awaiting only 2, those beginning with big "F":
ogrinfo fehlertest.csv -dialect SQLite -sql "select * from fehlertest where rstatus like 'F%'"

This query finds nothing:
ogrinfo fehlertest.csv -dialect SQLite -sql "select * from fehlertest where rstatus like '%F'"

**Comment by rouault** [comment]
@dpalmetzvbg I don't reproduce on my Linux install starting with GDAL 2.2.
```
$ ogrinfo Export.shp -sql "select COUNT(*) as Anzahl from Export WHERE rstatus like '%Fehler'" -dialect SQLITE -q

Layer name: SELECT
OGRFeature(SELECT):0
  Anzahl (Integer) = 2
```
 But with GDAL 2.1 I get a 
```
ERROR 1: In ExecuteSQL(): sqlite3_step(select * from Export WHERE rstatus LIKE '%Fehler'):
  SQL logic error or missing database
```
so this has been fixed in GDAL 2.2. Much likely with https://trac.osgeo.org/gdal/ticket/7149

@jratike80 https://www.sqlite.org/lang_expr.html mentions that LIKE is case insensitive.
and a '%F' query would match only text ending with 'f' or 'F'

so everything is logical. Closing
